src/graphql/graphql.ts

```diff
@@ -427,6 +435,126 @@ function createOperationDefinition(
   };
 }
 
+/**
+ * Paginate v2 queries with where clause on id
+ */
+export function paginateWithWhereV2(query: string): PaginatedQuery {
+  const edgesPath: string[] = [];
+  const ast = gql.visit(gql.parse(query), {
+    Document(node) {
+      if (node.definitions.length !== 1) {
+        throw invalidQuery(
+          'document should contain a single query operation definition'
+        );
+      }
+    },
+    OperationDefinition(node) {
+      if (node.operation !== 'query') {
+        throw invalidQuery('only query operations are supported');
+      }
+
+      // Add pagination variables to query operation
+      return createOperationDefinition(node, [
+        ['id', 'String'],
+        ['limit', 'Int'],
+      ]);
+    },
+    Field: {
+      enter(node) {
+        if (edgesPath.length) {
+          // Skip rest of nodes once edges path has been set
+          return false;
+        }
+        edgesPath.push(node.name.value);
+        const whereArg = node.arguments?.find(
+          (n) => n.name.value === 'where'
+        )?.value;
+        return {
+          ...node,
+          arguments: [
```

Comment:
so do you overwrite the existing arguments completely?

Reply:
I only keep existing where arg fields

Reply:
if we want to merge where predicates, we can lift [this code](https://github.com/faros-ai/clio/blob/master/server-v2/src/hasura-client.ts#L149) out of clio